src/opensynth/models/faraday/losses.py

```diff
@@ -59,12 +82,44 @@ def quantile_loss(
     Returns:
         torch.Tensor: Quantile loss
     """
-    return torch.mean(
-        torch.max(
-            quantile * (y_real - y_pred), (1 - quantile) * (y_pred - y_real)
-        )
+
+    quantile_loss = torch.max(
+        quantile * (y_real - y_pred), (1 - quantile) * (y_pred - y_real)
     )
 
+    if sample_weights is not None:
+        quantile_loss = quantile_loss * sample_weights
```

Comment:
I feel very pedantic writing this, but I think it avoids a copy of `quantile_loss`. Will leave it up to you if you want to change it. 
```suggestion
        quantile_loss *= sample_weights
```

Reply:
Haha i'm the opposite, I always do x = x + y and avoid *= because of accidentally overriding other assigned objects in place. E.g.

`*= operator`
```python
a = torch.tensor([1,2,3])
b = a
a *= 2 

--> a is now torch.tensor([2,4,6])
--> but b is also now torch.tensor([2,4,6]) because b=a doesn't result in a new copy of `b` variable but share the same memory
```

`x = x * 2 explicitly`
```python
a = torch.tensor([1,2,3])
b = a
a = a * 2

--> a is torch.tensor([2,4,6])
---> b however remains torch.tensor([1,2,3])
```

I think in this case, everything is scoped within a function and there isn't any other shared objects, so it's safe, but got bitten by this weird behaviour before and have always stayed away from `*=` haha.